### DISCUSSION TITLE
MODPATH create_mp7 with mpbas

### DISCUSSION THREAD

**Discussion by WaterqualityFloris** [modflowpy/flopy#2339] [Q&A] [answered]
Hi all,

I have this flopy code snippet that sets up a modpath simulation. I add a porosity to the simulation through an array called ' porosity' with Modpath7Bas, after setting up the simulation with create_mp7

```
mp = flopy.modpath.Modpath7.create_mp7(
    modelname=mpnamb,
    trackdir="backward",
    flowmodel=gwf,
    model_ws=ws,
    rowcelldivisions=1,
    columncelldivisions=1,
    layercelldivisions=1, create basic backward tracking modpath simulation
    exe_name='mp7',
    verbose=True
)

mpbas = flopy.modpath.Modpath7Bas(mp, porosity=porosity)

mp.write_input()
mp.run_model(silent=True, report=True)
```
However, when I run this code, I get this error:

flopy/mbase.py:668: UserWarning: Two packages of the same type, Replacing existing 'MPBAS' package.
  warn(

This results from me using Modpath7Bas. I did not expect this warning and wasn't able to find a good example on how to otherwise set up this problem. It seems that it does not affect calculation results, which have an output that I would expect with changing porosity in my model. However, should I be worried about this warning and are there ways to work around it? If you have a good advise or a relevant section in the documentation or in these discussions that I missed, I'd be glad to get that from you! Looking forward to hearing your advice,

Floris

**Comment by martclanor** [comment, marked as answer]
Hi Floris, 

I think it's because `Modpath7.create_mp7` creates already a `Modpath7Bas` object internally. Perhaps a possible solution is to copy the implementation inside `create_mp7` (as linked below) to your script, then passing `porosity` to the `Modpath7Bas` object

https://github.com/modflowpy/flopy/blob/f96d173c4f82410f4cd35489fc293217df243a43/flopy/modpath/mp7.py#L463-L509

Maybe it's a good idea to include `porosity` as a parameter of `create_mp7`, I'll try to submit a PR on this.
